Develop

bobtheragelot opened the pull request:
pretty much everything

SHS-Programmers committed:
Delete duplicate.

SHS-Programmers committed:
Revert "Delete duplicate."

This reverts commit 5d94981e

SHS-Programmers committed:
Copied Ryan's unfinished feeder code and put it in its own branch.

SHS-Programmers committed:
Added untested shooter constants

SHS-Programmers committed:
Added Neo dependencies.

SHS-Programmers committed:
Renamed Feeder to Magazine; added initial commands.

SHS-Programmers committed:
Add pneumatics code.
 - Drivetrain shifting.
 - Run compressor.

SHS-Programmers committed:
Fix code.

SHS-Programmers committed:
Initial Mast subsystems and commands

SHS-Programmers committed:
Fix code.

SHS-Programmers committed:
Merge pull request #35 from Staley-Robotics/pneumatics

Pneumatics
ok, this is epic.

SHS-Programmers committed:
Added SpinToColor and SpinToCount for the WOF

SHS-Programmers committed:
Added pistonHardStop. Updated to Dustin's suggestions

SHS-Programmers committed:
Added pistonHardStop. Updated to Dustin's suggestions

SHS-Programmers committed:
Initial mast (#30)

* Add initial telescopic mast code

* Update initial-vision code

* Initial Winch subsystems and commands.

* Initial Mast subsystems and commands. Also accidentally rewrote winch here please dustin don't hurt

* Updated to dustin's standards

* Updated to dustin's standards for real this time

* Added SpinToColor and SpinToCount for the WOF

Co-authored-by: Dustin Klein <dbklein4@gmail.com>

Dustin-Klein committed:
Merge pull request #31 from Staley-Robotics/initial-feeder

Initial Magazine

BEF-BEF committed:
Merge branch 'develop' into initial-shooter

Dustin-Klein committed:
Merge pull request #32 from Staley-Robotics/initial-shooter

Initial shooter

SHS-Programmers committed:
Updated WOF to Dustin's suggestions

SHS-Programmers committed:
Merge branch 'develop' into initial-WOF

SHS-Programmers committed:
Merge pull request #38 from Staley-Robotics/initial-WOF

Initial wof

SHS-Programmers committed:
Fixed checkstyle-suppressions.xml

SHS-Programmers committed:
Added VisionYawAlign

SHS-Programmers committed:
Added ShootBalls (shoots balls until timeout)

SHS-Programmers committed:
Added ShootBalls (shoots balls until timeout)

SHS-Programmers committed:
Merge pull request #47 from Staley-Robotics/additionalCommands

Additional commands
ebic

SHS-Programmers committed:
Make Autos.

SHS-Programmers committed:
Merge branch 'develop' of https://github.com/Staley-Robotics/StaleyRC2020 into develop

# Conflicts:
#	src/main/java/frc/robot/RobotContainer.java

SHS-Programmers committed:
Update gitignore.
 - Remove PathWeaver and PathFinder files

SHS-Programmers committed:
Resolved #46 and #41. Added error exceptions for motor controller initializations. removed unused imports.

SHS-Programmers committed:
Rename Autos.
Add comments for Autos.

SHS-Programmers committed:
Resolved #46 and #41. Added error exceptions for motor controller initializations. removed unused imports.

BEF-BEF committed:
Update Intake.java

SHS-Programmers committed:
Update checkstyle-suppressions.xml

SHS-Programmers committed:
Merge branch 'fuvk' of https://github.com/Staley-Robotics/StaleyRC2020 into develop

# Conflicts:
#	config/checkstyle/checkstyle-suppressions.xml
#	src/main/java/frc/robot/RobotContainer.java
#	src/main/java/frc/robot/commands/auto/AutoBrettV7.java
#	src/main/java/frc/robot/commands/auto/CenteredSixBall.java
#	src/main/java/frc/robot/commands/auto/ShootThenMoveOff.java

SHS-Programmers committed:
Conform to Checkstyle.

SHS-Programmers committed:
Bound commands to keys.

SHS-Programmers committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	config/checkstyle/checkstyle-suppressions.xml
#	src/main/java/frc/robot/subsystems/WallOfFlesh.java

SHS-Programmers committed:
Bound commands to keys.

SHS-Programmers committed:
Created docs directory and moved documents into it.

SHS-Programmers committed:
Created Auto documentation.

SHS-Programmers committed:
add closed loop shooting. probably don't need to count balls i'm bad.

SHS-Programmers committed:
Fixed Autos, created encoder stopping point for joint.

Dustin-Klein committed:
Merge pull request #55 from Staley-Robotics/doc-cleanup

Doc cleanup

SHS-Programmers committed:
Updated flywheel threshold logic.

SHS-Programmers committed:
Merge pull request #54 from Staley-Robotics/ShooterLogic

Adds closed loop shooting logic

SHS-Programmers committed:
Fixed Autos, created encoder stopping point for joint.

SHS-Programmers committed:
Remove unnecessary comments.

Dustin-Klein committed:
Merge branch 'develop' into fix-the-autos

Dustin-Klein committed:
Merge pull request #56 from Staley-Robotics/fix-the-autos

Fix the autos

SHS-Programmers committed:
Fixed checkstyle.

SHS-Programmers committed:
Added controller map.

Dustin-Klein committed:
Merge pull request #57 from Staley-Robotics/controller-map

Added controller map.

SHS-Programmers committed:
Added run magazine to the run intake button.

SHS-Programmers committed:
Added a way to run the intake backwards by pressing the back button. Updated the controller map.

SHS-Programmers committed:
Merge pull request #64 from Staley-Robotics/run-intake-backwards

Run intake backwards

SHS-Programmers committed:
Makes singletons' constructors private.

SHS-Programmers committed:
Merge pull request #65 from Staley-Robotics/fix-the-constructors

Makes singletons' constructors private.

SHS-Programmers committed:
Changed the implementation of toggle joint in autos to use PID control.

SHS-Programmers committed:
Removed timeout.

SHS-Programmers committed:
Added shifting logic.

SHS-Programmers committed:
Merge pull request #66 from Staley-Robotics/fix-intake-toggle

Changed the implementation of toggle joint in autos to use PID control.

SHS-Programmers committed:
Added shifting logic.

Dustin-Klein committed:
Merge branch 'develop' into auto-shift

SHS-Programmers committed:
Added shifting logic.

Dustin-Klein committed:
Merge pull request #68 from Staley-Robotics/auto-shift

Added shifting logic.

SHS-Programmers committed:
Added ability to read WOF color from driver station.

SHS-Programmers committed:
Updated Structure.md.

SHS-Programmers committed:
Adds vision yaw align where required for existing auto modes

Dustin-Klein committed:
Merge pull request #72 from Staley-Robotics/auto-vision-fix

Adds vision yaw align where required for existing auto modes

SHS-Programmers committed:
Fixed mistakes.

SHS-Programmers committed:
Merge pull request #71 from Staley-Robotics/update-structureMD

Updated Structure.md.

bobtheragelot commented on config/checkstyle/checkstyle-suppressions.xml:
```diff
@@ -5,22 +5,17 @@
-
   <suppress files="DriveTrain.java"
-    lines="0-319"
+    lines="0-470"
```
bruh

bobtheragelot commented on config/checkstyle/checkstyle-suppressions.xml:
```diff
@@ -35,9 +30,66 @@
   <suppress files="TestReverse.java"
     lines="20"
     message="Missing a Javadoc comment."/>
+  <suppress files="WallOfFlesh.java"
```
group check style by file name. then group check style alphabetically by file name

bobtheragelot replied:
then reorder lines within file name by line number

bobtheragelot commented on docs/AutoModes.md:
```diff
@@ -0,0 +1,84 @@
+
+##Centered Six Ball 
+
+**Inner 24 Points**
```
backwards. add outline points.

bobtheragelot commented on docs/AutoModes.md:
```diff
@@ -0,0 +1,84 @@
+#Auto Modes
+The five different autos, points scored, and their steps.
+
```
make a note of what inner and outer mean. define inner = 6 points and outer = 4

bobtheragelot commented on docs/AutoModes.md:
```diff
@@ -0,0 +1,84 @@
+
+**Outer 36 Points**
+
+This auto shoots all it's stored balls, then drives to the trench run and picks up three more. After picking up the balls it returns, and shoots them.
```
its

bobtheragelot commented on docs/AutoModes.md:
```diff
@@ -0,0 +1,84 @@
+
+6. Turns a sharp right turn back around to the Power Port.
+
+7. Drives forward 2.5 meters and shoots.
```
shoots how many balls ideally assuming 0 misses

bobtheragelot commented on docs/AutoModes.md:
```diff
@@ -0,0 +1,84 @@
+
+##Left Six Ball 
+
+**Inner 24 Points**
```
reversed

bobtheragelot commented on docs/AutoModes.md:
```diff
@@ -0,0 +1,84 @@
+
+6. Makes a sharp right turn to face power port.
+
+7. Drives 4 meters forward, and then shoots.
```
how many balls

bobtheragelot commented on docs/AutoModes.md:
```diff
@@ -0,0 +1,84 @@
+
+4. Drives in reverse out of the trench.
+
+5. Turns around and shoots.
```
how many balls

bobtheragelot commented on docs/ControllerMap.md:
```diff
@@ -0,0 +1,43 @@
+
+**B**: The B button lines the robot up to shoot.
+
+**Left Joystick**: The left joystick controls turning. If it is moved left past the deadzone, the robot turns left. The same is true for right.
```
x axis

SHS-Programmers replied:
No

bobtheragelot commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -60,12 +60,16 @@
-  public static final class OperatorInputConstants {
+    public static final double turnP = 0.5;
+    public static final double turnI = 0.5;
+    public static final double turnD = 0.5;
```
dustin bad but also don't 0.5. have p be like 0.1 then I and d be 0 for now.

bobtheragelot commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -74,11 +78,64 @@
 
     public static final double defaultIntakePower = 0.2;
-    public static final double defualtJointPower = 0.2;
+    public static final double defaultJointPower = 0.2;
```
do not need anymore

bobtheragelot commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -41,25 +55,29 @@
   private XboxController altController;
 
```
declare + initialize every subsystem here

PoisonPhang commented on src/main/java/frc/robot/commands/drivetrain/TurnToAngle.java:
```diff
@@ -0,0 +1,49 @@
+        // Set reference to target
+        targetAngleDegrees,
+        // Pipe output to turn robot
+        output -> driveTrain.worldOfTanksDrive(0, 0, output),
```
Unsure if this is necessary, but you use DriveTrain.getInstance in every other auto when you need the DriveTrain object. You might want to use that method here to keep consistency. You may want to create a class local DriveTrain object to store this instance in instead.
```suggestion
        output -> DriveTrain.getInstance.worldOfTanksDrive(0, 0, output),
```

bobtheragelot commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -83,15 +104,53 @@ private void configureButtonBindings() {
+    wofDown.whenPressed(wallOfFlesh::lowerWof, wallOfFlesh);
+
+    JoystickButton wofSpinNumber = new JoystickButton(altController, DpadButton.kDpadLeft.value);
+    wofSpinNumber.whenPressed(new SpinToCount(3.5));
```
constants. wofconstants goalSpinCount = 3.5

bobtheragelot commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -83,15 +104,53 @@ private void configureButtonBindings() {
+    wofSpinNumber.whenPressed(new SpinToCount(3.5));
+
+    JoystickButton wofSpinColor = new JoystickButton(altController, DpadButton.kDpadRight.value);
+    wofSpinColor.whenPressed(new SpinToColor(Color.kBlue));
```
wof.targetColor reads this from field. Must convert string to color as targetColor is a string. kBlue is for testing.

bobtheragelot commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -83,15 +104,53 @@ private void configureButtonBindings() {
+    wofSpinColor.whenPressed(new SpinToColor(Color.kBlue));
+
+    JoystickButton mastUp = new JoystickButton(altController, Button.kBumperLeft.value);
+    mastUp.whileHeld(new RunMast(0.5));
```
use a constant

bobtheragelot commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -83,15 +104,53 @@ private void configureButtonBindings() {
+    mastUp.whileHeld(new RunMast(0.5));
+
+    JoystickButton mastDown = new JoystickButton(altController, Button.kBumperRight.value);
+    mastDown.whileHeld(new RunMast(-0.5));
```
use a constant

bobtheragelot commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -83,15 +104,53 @@ private void configureButtonBindings() {
+    mastDown.whileHeld(new RunMast(-0.5));
+
+    JoystickButton winchExtend = new JoystickButton(altController, Axis.kRightTrigger.value);
+    winchExtend.whileHeld(new RunWinch(0.5));
```
use a constant

bobtheragelot commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -83,15 +104,53 @@ private void configureButtonBindings() {
+    winchExtend.whileHeld(new RunWinch(0.5));
+
+    JoystickButton winchRetract = new JoystickButton(altController, Axis.kLeftTrigger.value);
+    winchRetract.whileHeld(new RunWinch(-0.5));
```
use a constant

bobtheragelot commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -83,15 +104,53 @@ private void configureButtonBindings() {
 
-    JoystickButton toggleJointPosition = new JoystickButton(altController, Button.kY.value);
-    toggleJointPosition.whenPressed(new ToggleJoint(defualtJointPower).withTimeout(1));
+    JoystickButton togglePiston = new JoystickButton(driveController, Button.kX.value);
```
toggle what piston. make better name

PoisonPhang commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -0,0 +1,160 @@
+    PIDController = rightShooterNeo.getPIDController();
+    PIDController.setFeedbackDevice(shooterEncoder);
+
+    targetSpeed = 0;
```
Is it necessary to set this to 0 in the constructor vs along side declaration like the other target values?

bobtheragelot replied:
yes

PoisonPhang reviewed (dismissed):
Check my comments for minor changes that you may want to make. Otherwise looks good. +1

bobtheragelot commented on src/main/java/frc/robot/commands/auto/AutoBrettV7.java:
```diff
@@ -3,35 +3,21 @@
 
 /**
- * Drives Forward 1.5 meters, waits 3 seconds and travels another 1.5 meters.
+ * Moves past Auto line.
```
Moves off of auto line.

bobtheragelot commented on src/main/java/frc/robot/commands/auto/AutoBrettV7.java:
```diff
@@ -3,35 +3,21 @@
-        drive.getPoseListFromPathWeaverJson("ForwardContinue"),
-        drive.getTrajectoryConfig(false));
+    Trajectory forwardPastAutoLine = TrajectoryGenerator.generateTrajectory(
+        driveTrain.getPoseListFromPathWeaverJson("Forward"),
```
autobrett should move the robot away from wall

Dustin-Klein replied:
etahn do this

bobtheragelot commented on src/main/java/frc/robot/commands/auto/CenteredSixBall.java:
```diff
@@ -0,0 +1,46 @@
+ * Immediately shoots pre-loaded balls. Moves to friendly trench to gather 3 balls. Turns and moves
+ * back to face the generator and then shoots the 3 collected balls.
+ */
+public class CenteredSixBall extends LowGearAuto {
```
Change all auto names.

Dustin-Klein replied:
etahn do this

bobtheragelot commented on src/main/java/frc/robot/commands/auto/CenteredSixBall.java:
```diff
@@ -0,0 +1,46 @@
+        new InstantCommand(driveTrain::resetOdometry, driveTrain),
+        new InstantCommand(driveTrain::zeroEncoder, driveTrain),
+        new VisionYawAlign(),
+        new ShootBallsOpenLoop(vision.calculateDistance(vision.getPitch())),
```
use closed loop in all autos. add logic in closed loop to automatically calculate threshold based off of distance.

Dustin-Klein replied:
closed loop in all autos. auto threshold maybe later.

bobtheragelot commented on src/main/java/frc/robot/commands/auto/CenteredSixBall.java:
```diff
@@ -0,0 +1,46 @@
+        new ShootBallsOpenLoop(vision.calculateDistance(vision.getPitch())),
+        driveTrain.getAutonomousCommandFromTrajectory(trajectoryForward),
+        new ToggleJoint(),
+        new RunIntake(defaultIntakePower).withTimeout(10),
```
adjust 10 to something better

bobtheragelot commented on src/main/java/frc/robot/commands/auto/CenteredSixBall.java:
```diff
@@ -0,0 +1,46 @@
+        new ShootBallsOpenLoop(vision.calculateDistance(vision.getPitch())),
+        driveTrain.getAutonomousCommandFromTrajectory(trajectoryForward),
+        new ToggleJoint(),
+        new RunIntake(defaultIntakePower).withTimeout(10),
```
run intake parallel. it is sequential right now.

Dustin-Klein replied:
etahn do this

bobtheragelot commented on src/main/java/frc/robot/commands/auto/RightToEnemyTrenchToShoot.java:
```diff
@@ -0,0 +1,44 @@
+
+    Trajectory trajectoryForward = TrajectoryGenerator.generateTrajectory(
+        driveTrain.getPoseListFromPathWeaverJson("ForwardToTrench"),
+        driveTrain.getTrajectoryConfig(false));
```
create trajectory config

bobtheragelot replied:
rename trajectories

Dustin-Klein replied:
etahn do this

bobtheragelot commented on src/main/java/frc/robot/commands/intake/ToggleJoint.java:
```diff
@@ -7,51 +7,37 @@
+    }
+
+    if (intake.getPivotState() == PivotState.up) {
+      intake.setPivotState(PivotState.down);
```
make toggle joint set the pivot state to what it is setting the guy to

bobtheragelot commented on src/main/java/frc/robot/commands/intake/ToggleJoint.java:
```diff
@@ -7,51 +7,37 @@
 public class ToggleJoint extends CommandBase {
```
delete command. replace with lamb calling intake.toggleJoint()

toggle joint(){
pivot state = not pivot state
pid.set = pivotstategoalencoder
}

SHS-Programmers replied:
No

bobtheragelot commented on src/main/java/frc/robot/commands/magazine/RunMagazine.java:
```diff
@@ -0,0 +1,34 @@
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Magazine;
+
+public class RunMagazine extends CommandBase {
```
delete. replace with lamb

Dustin-Klein replied:
no

bobtheragelot commented on src/main/java/frc/robot/commands/mast/RunMast.java:
```diff
@@ -0,0 +1,37 @@
+package frc.robot.commands.mast;
```
delete. replace with lamb

bobtheragelot commented on src/main/java/frc/robot/commands/shooter/ShootBallsClosedLoop.java:
```diff
@@ -0,0 +1,56 @@
+   *
+   * @param distance distance to base shot off of.
+   */
+  public ShootBallsClosedLoop(double distance, double percentSpeedRequired) {
```
override shootBallsClosedLoop(distance, percentSpeedRequired) with shootBallsClosedLoop(double distance) that automatically calculates threshold based on distance.

Dustin-Klein replied:
maybe later

bobtheragelot commented on src/main/java/frc/robot/commands/shooter/ShootBallsClosedLoop.java:
```diff
@@ -0,0 +1,56 @@
+  @Override
+  public void initialize() {
+    shooter.setFlyWheelSpeed(shooter.calculateSurfaceVelocity(distance));
+    magazine.retractHardStop();
```
run magazine backwards real quick so that it doesn't get caught on the piston

Dustin-Klein replied:
dustin says don't need probably maybe

bobtheragelot commented on src/main/java/frc/robot/commands/shooter/ShootBallsClosedLoop.java:
```diff
@@ -0,0 +1,56 @@
+
+    double percentage = flyWheelSpeed / targetSpeed;
+    if (Math.abs(1 - percentage) <= Math.abs(1 - percentSpeedRequired)) {
+      magazine.runMagazine(0.5);
```
replace 0.5 with constant

bobtheragelot commented on src/main/java/frc/robot/commands/shooter/ShootBallsOpenLoop.java:
```diff
@@ -0,0 +1,47 @@
+package frc.robot.commands.shooter;
+
```
delete this. if you want open loop call closed loop with a 0 threshold.

bobtheragelot commented on src/main/java/frc/robot/commands/winch/RunWinch.java:
```diff
@@ -0,0 +1,44 @@
+package frc.robot.commands.winch;
```
delete. replace with lamb

bobtheragelot commented on src/main/java/frc/robot/commands/wof/SpinToColor.java:
```diff
@@ -0,0 +1,58 @@
+
+  private Color findColor;
+
+  /**
```
suppress

bobtheragelot commented on src/main/java/frc/robot/commands/wof/SpinToColor.java:
```diff
@@ -0,0 +1,58 @@
+
+  @Override
+  public void execute() {
+    //while not reading color that would make the field's sensor read what we want run motor
```
delete

SHS-Programmers replied:
From here down I am the lord, and I will make the changes

bobtheragelot commented on src/main/java/frc/robot/commands/wof/SpinToCount.java:
```diff
@@ -0,0 +1,39 @@
+  @Override
+  public void execute() {
+    wallOfFlesh
+        .spinDistance(spinCount * wallOfFleshCircumference / wallOfFleshSpinnerCircumference);
```
use a conversion method rather than doing inline

bobtheragelot commented on src/main/java/frc/robot/subsystems/DriveTrain.java:
```diff
@@ -116,11 +134,18 @@ public DriveTrain() {
 
     odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getHeading()));
 
+    shifter = new DoubleSolenoid(0, 7);
```
replace with constants. "use an array with 2 values for double solenoid's ports because better "dustin

bobtheragelot commented on src/main/java/frc/robot/subsystems/DriveTrain.java:
```diff
@@ -136,6 +161,8 @@ public void periodic() {
     SmartDashboard.putNumber("RightEncoder(m): ", stepsToMeters(getRightEncoderPosition()));
 
     SmartDashboard.putNumber("Heading: ", getHeading());
```
heading is yaw but I don't know what heading is until I think heading is yaw.

bobtheragelot commented on src/main/java/frc/robot/subsystems/DriveTrain.java:
```diff
@@ -162,8 +189,8 @@ public void worldOfTanksDrive(double forward, double backward, double rotate) {
     // Deadzones for rotate.
-    if (rotate > 0.1 || rotate < 0.1) {
-      rotate = rotate * turnSpeedModifier;
+    if (rotate > rotateDeadzone || rotate < rotateDeadzone) {
```
look at shooter threshold to do this better

bobtheragelot committed:
Update src/main/java/frc/robot/commands/drivetrain/TurnToAngle.java

Co-Authored-By: Connor Davis <poisonphang@gmail.com>

bobtheragelot commented on src/main/java/frc/robot/commands/drivetrain/TurnToAngle.java:
```diff
@@ -0,0 +1,49 @@
+        // Set reference to target
+        targetAngleDegrees,
+        // Pipe output to turn robot
+        output -> DriveTrain.getInstance.worldOfTanksDrive(0, 0, output),
```
make method

bobtheragelot commented on src/main/java/frc/robot/commands/drivetrain/TurnToAngle.java:
```diff
@@ -0,0 +1,49 @@
+        // Pipe output to turn robot
+        output -> DriveTrain.getInstance.worldOfTanksDrive(0, 0, output),
+        // Require the drive
+        driveTrain);
```
DriveTrain.getInstance() instead of driveTrain parameter

bobtheragelot commented on src/main/java/frc/robot/subsystems/DriveTrain.java:
```diff
@@ -402,4 +446,36 @@ public Command getAutonomousCommandFromTrajectory(Trajectory trajectory) {
+  }
+
+  private boolean isLowGearOptimal() {
+    if ((getLeftEncoderMetersPerSecondVelocity() + getRightEncoderMetersPerSecondVelocity()) / 2
```
if Math.abs(average) < shiftPointsMetersPerSecond

also
if difference between left side and right side velocity > 20% don't shift. that prevents you from shifting into low gear while turning.
or
do we want it to shift to high automatically when turning?

bobtheragelot commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -7,69 +7,97 @@
+import static frc.robot.Constants.IntakeConstants.higherPosition;
 import static frc.robot.Constants.IntakeConstants.intakeMotorPort;
 import static frc.robot.Constants.IntakeConstants.jointMotorPort;
-import static frc.robot.Constants.SensorConstants.limitSwitchPort;
```
add limit switch at top so that we don't crush electronics

bobtheragelot commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -7,69 +7,97 @@
+    talonConfig.slot0.kI = 0.0;
+    talonConfig.slot0.kD = kD;
+    talonConfig.slot0.integralZone = 400;
+    talonConfig.slot0.closedLoopPeakOutput = 1.0;
```
do we want to let this be maxed? maxed is a lot

bobtheragelot commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -7,69 +7,97 @@
-  public void runJoint(double power) {
-    jointMotor.set(power);
+  public void lowerIntake() {
+    jointMotor.set(ControlMode.Position, lowerPosition);
```
set pivot state enum

bobtheragelot commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -7,69 +7,97 @@
-  public void runIntake(double power) {
-    intakeMotor.set(power);
+  public void raiseIntake() {
+    jointMotor.set(ControlMode.Position, higherPosition);
```
set pivot state enum

bobtheragelot commented on src/main/java/frc/robot/subsystems/Magazine.java:
```diff
@@ -0,0 +1,89 @@
+  }
+
+  /**
+   * Constructor.
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/Magazine.java:
```diff
@@ -0,0 +1,89 @@
+  }
+
+  /**
+   * Makes Magazine a singleton.
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/Magazine.java:
```diff
@@ -0,0 +1,89 @@
+    return instance;
+  }
+
+  public void extendHardStop() {
```
this is what lowerIntake should look like

bobtheragelot commented on src/main/java/frc/robot/subsystems/Mast.java:
```diff
@@ -0,0 +1,58 @@
+    } catch (RuntimeException ex) {
+      DriverStation
+          .reportError("Error Instantiating Mast Motor Controllers: " + ex.getMessage(), true);
+    }
```
victor sp don't error

bobtheragelot commented on src/main/java/frc/robot/subsystems/Mast.java:
```diff
@@ -0,0 +1,58 @@
+    }
+  }
+
+  /**
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/Pneumatics.java:
```diff
@@ -0,0 +1,46 @@
+
+/**
+ * Runs compressor.
+ */
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/Pneumatics.java:
```diff
@@ -0,0 +1,46 @@
+
+  /**
+   * Makes Pneumatics a singleton.
+   */
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -0,0 +1,160 @@
+
+  private CANEncoder shooterEncoder;
+
+  /*These thresholds are used by commands to decide how close the flywhell's velocity needs to be to
```
not a javadoc

bobtheragelot commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -0,0 +1,160 @@
+
+  //target height and shooter height in meters
+  public final double targetHeight = 5;
+  public final double shooterHeight = 4;
```
make constants. delete accurateshootvelocity and fastshootvelocity

bobtheragelot commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -0,0 +1,160 @@
+   * Makes shooter a singleton.
+   *
+   * @return the instance of magazine
+   */
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -0,0 +1,160 @@
+   *
+   * @param distance distance from the target in inches.
+   */
+  public double calculateSurfaceVelocity(double distance) {
+    //Lots of commented math I don't want to copy
+    return 2;
+  }
```
don't return 2 actually do the stuff.

SHS-Programmers replied:
No. Later

bobtheragelot commented on src/main/java/frc/robot/subsystems/Vision.java:
```diff
@@ -27,6 +27,7 @@ private Vision() {
    * Makes Vision a singleton.
+   *
    * @return Vision
    */
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/Vision.java:
```diff
@@ -71,7 +72,7 @@ public boolean tapeDetected() {
+   * @return Calculated Distance.
    */
   public double calculateDistance(double pitch) {
     return (bottomOfTargetHeight - fixedCameraHeight) / (Math
```
calculate distance() no params.

bobtheragelot commented on src/main/java/frc/robot/subsystems/WallOfFlesh.java:
```diff
@@ -7,18 +7,124 @@
+
+  /**
+   * Constructor.
+   */
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/WallOfFlesh.java:
```diff
@@ -7,18 +7,124 @@
+      DriverStation
+          .reportError("Error Instantiating WOF Motor Controllers: " + ex.getMessage(), true);
+    }
+    wofPiston = new DoubleSolenoid(0, 7);
```
constants

bobtheragelot commented on src/main/java/frc/robot/subsystems/WallOfFlesh.java:
```diff
@@ -7,18 +7,124 @@
+   * Makes WOF a singleton.
+   *
+   * @return WOF instance.
    */
```
suppress

bobtheragelot commented on src/main/java/frc/robot/subsystems/WallOfFlesh.java:
```diff
@@ -7,18 +7,124 @@
+  public void readDriveStationMessage() {
+    String gameData = DriverStation.getInstance().getGameSpecificMessage();
+    if (gameData.length() > 0) {
+      targetRecieved(gameData.charAt(0));
```
have goalColor be a color not a string

bobtheragelot commented on src/main/java/frc/robot/subsystems/Winch.java:
```diff
@@ -7,18 +7,57 @@
+   * Makes Winch a singleton.
+   *
+   * @return Winch instance
+   */
```
suppress

SHS-Programmers committed:
checkstyle + build running

SHS-Programmers committed:
Merge pull request #79 from Staley-Robotics/develop-fix

Develop fix

ebic

SHS-Programmers committed:
organized checkstyle

SHS-Programmers committed:
Fix docs.

SHS-Programmers committed:
Fix docs.

SHS-Programmers committed:
Fix docs.

SHS-Programmers committed:
Enhance docs.

SHS-Programmers committed:
Enhance docs.

Dustin-Klein committed:
Merge pull request #81 from Staley-Robotics/ethan-devfix

Ethan devfix

SHS-Programmers committed:
Change DriveTrain.
 - Add ShifterPorts array.
 - Make worldOfTanksDrive deadzone logic more efficient
 - Add isLowGearOptimal absolute value
 - Add isLowGearOptimal turn logic *UNTESTED AND MAYBE BAD*

SHS-Programmers committed:
Update Intake and ToggleJoint.
 - Add LimitSwitch.

SHS-Programmers committed:
Update Intake and ToggleJoint.
 - Reduce closedLoopPeakOutput.

SHS-Programmers committed:
deleted open loop shooting. made winch a lamb.

SHS-Programmers committed:
Change WallOfFlesh
 - Add wallOfFleshSolenoids array.
 - Change colorTarget to Color instead of String

SHS-Programmers committed:
Change Winch
 - Remove unnecessary comment

SHS-Programmers committed:
Remove unnecessary WallOfFlesh import.
Reverse AutoBrettV7 Trajectory so robot moves away from wall backwards.

SHS-Programmers committed:
Rename Auto.
Change AutoModes.md to fit new names

SHS-Programmers committed:
changed ToggleJoint.

SHS-Programmers committed:
initializes all subsystems in RobotContainer

SHS-Programmers committed:
Add getColorTarget.
Rename getTrajectoryConfig.
Rename auto Trajectories.

SHS-Programmers committed:
Many Changes.

SHS-Programmers committed:
Many Changes.

SHS-Programmers committed:
Merge pull request #84 from Staley-Robotics/ethan-devfix

Ethan devfix
ok, this is epic

SHS-Programmers committed:
fixed checkstyle

SHS-Programmers committed:
Merge branch 'develop-fix' into develop

# Conflicts:
#	config/checkstyle/checkstyle-suppressions.xml
#	src/main/java/frc/robot/Constants.java
#	src/main/java/frc/robot/commands/auto/CentSixBall.java
#	src/main/java/frc/robot/commands/auto/LeftSixBall.java
#	src/main/java/frc/robot/commands/auto/ShootThenMoveOff.java
#	src/main/java/frc/robot/commands/auto/StealThenShoot.java
#	src/main/java/frc/robot/commands/intake/ToggleJoint.java
#	src/main/java/frc/robot/commands/shooter/ShootBallsCommandGroup.java
#	src/main/java/frc/robot/commands/wof/SpinToColor.java

SHS-Programmers committed:
Added some Shooter testing logic. Reorganized checkstyle-suppressions.xml

SHS-Programmers committed:
deleted extra method

SHS-Programmers committed:
Merge pull request #85 from Staley-Robotics/ShooterPlanning

Shooter planning

SHS-Programmers committed:
Updated AutoModes.md.

SHS-Programmers committed:
Fixed mistakes

SHS-Programmers committed:
Merge pull request #86 from Staley-Robotics/update-automode-doc

Updated AutoModes.md.

SHS-Programmers committed:
Fixed paths.
